osquery/distributed/distributed.cpp

```diff
@@ -66,9 +78,17 @@ Status Distributed::pullUpdates() {
     return Status(1, "Missing distributed plugin: " + distributed_plugin);
   }
 
+  // should not have any work in progress when this is called.
+  assert(results_.size() == 0);
```

Comment:
I'm generally ok with having asserts in production code but @muffins @fmanco @akindyakov